src/pdp/server.ts

```diff
@@ -515,6 +515,31 @@ export class PDPServer {
     return await downloadAndValidateCommP(response, parsedCommP)
   }
 
+  /**
+   * Get proof set details from the PDP server
+   * @param proofSetId - The ID of the proof set to fetch
+   * @returns Promise that resolves with proof set data
+   */
+  async getProofSet (proofSetId: string): Promise<ProofsetData> {
+    const response = await fetch(`${this._apiEndpoint}/pdp/proof-sets/${proofSetId}`, {
+      method: 'GET',
+      headers: {
+        Accept: 'application/json'
+      }
+    })
+
+    if (response.status === 404) {
+      throw new Error(`Proof set not found: ${proofSetId}`)
+    }
+
+    if (!response.ok) {
+      const errorText = await response.text()
+      throw new Error(`Failed to fetch proof set: ${response.status} ${response.statusText} - ${errorText}`)
+    }
+
+    return await response.json() as ProofsetData
```

Comment:
Noticing this made me remember that we're not doing any validation. So have a look at https://github.com/FilOzone/synapse-sdk/pull/113 for validation for the other responses so far. I'd like you to add the same for this PR. You could either wait for that PR to merge, or `git rebase upstream rvagg/validate-server` this branch and it should fit fairly cleanly on it and then you can add your new set of functions for this.